src/components/modals/WalletListModal.tsx

```diff
@@ -136,14 +135,14 @@ export function WalletListModal(props: Props) {
   const handlePaymentMethodPress = useHandler((fiatAccountId: string) => () => {
     bridge.resolve({ type: 'wyre', fiatAccountId })
   })
-  const handleWalletListPress = useHandler((walletId: string, tokenId: EdgeTokenId, customAsset?: CustomAsset) => {
+  const handleWalletListPress = useHandler(async (walletId: string, tokenId: EdgeTokenId, customAsset?: CustomAsset) => {
     if (walletId === '') {
       handleCancel()
       showError(lstrings.network_alert_title)
     } else if (customAsset != null) {
       bridge.resolve({ type: 'custom', customAsset })
     } else {
-      const wallet = currencyWallets[walletId]
+      const wallet = await account.waitForCurrencyWallet(walletId)
```

Comment:
I don't see how the wallet could even appear in the list if it hasn't loaded already, but I guess this is safe either way.

Reply:
Has to do with newly created wallets not being ready yet before going on to get the `currencyCode` and resolve